### ISSUE TITLE
Simplify mbedtls_psa_rsa_load_representation and mbedtls_psa_ecp_load_representation

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#8776]
For historical reasons, `mbedtls_psa_rsa_load_representation` and `mbedtls_psa_ecp_load_representation` allocate a context on the heap, and their callers free that context. This happened because the allocation used to be inside a call to `mbedtls_pk_setup`. Since very long ago for ECP, and since https://github.com/Mbed-TLS/mbedtls/pull/8740 for RSA, the pk module is no longer involved, so the context can be allocated on the stack of the caller, which simplifies the code.

Goal of this issue: refactor so that `mbedtls_psa_rsa_load_representation` and `mbedtls_psa_ecp_load_representation` take a pointer-to-context rather than pointer-to-pointer-to-context, and their caller allocates the context of the stack. The existing testing should be sufficient.

**Comment by valeriosetti** [comment]
Can I add this issue to  the [Mbed TLS 3.6 Release](https://github.com/orgs/Mbed-TLS/projects/1#column-19402885) EPIC?
It would be in the same place as the issue that originated this one

**Comment by gilles-peskine-arm** [comment]
I don't think it belongs in the release epic because it's not release-critical. It's something we want to do in Q1 to avoid adding technical debt. That is probably the description of an epic, like “driver-only cleanups” but for work other than driver-only.

**Comment by mpg** [comment]
I'm not sure driver-only cleanups would be the right place: it contains things that are probably easier to address after 4.0, so it seems likely we might postpone the entire EPIC until after 4.0. This issue on the other hand can be done any time.

I understand your point that it's not release critical and it's clearly not a MUST for 4.0, but I also understand Valerio's temptation to just put it in the same EPIC as the PR that gave rise to it - after all, he could almost have addressed that in 8740 too.

I expect that while working on 4.0, we'll probably create or (re)discover a lot of opportunities for simplification or code size savings in existing code (be it PSA or legacy crypto modules), so we should probably have an EPIC for those.

**Comment by mpg** [comment]
See [this comment](https://github.com/Mbed-TLS/mbedtls/pull/8740#pullrequestreview-1854607581): 

> There are a few places where the refactoring allows further simplification. I'd like to do those before the next release (avoid acquiring technical debt) but we can do it in a follow-up pull request.

which in mind mind suggested "3.6 nice-to-have" for this issue.
